syntax = "proto3";

package assfire.api.v1.tsp;

import "assfire/api/v1/tsp/concepts.proto";

message WorkerTask {
  string task_id = 1;
  TspTask task = 2;
}

message WorkerControlSignal {
  enum Type {
    WORKER_CONTROL_SIGNAL_TYPE_UNKNOWN = 0;
    WORKER_CONTROL_SIGNAL_TYPE_PAUSE = 1;
    WORKER_CONTROL_SIGNAL_TYPE_INTERRUPT = 2;
  }

  string task_id = 1;
  Type signal_type = 2;
}

message TspSolutionResult {
  TspSolution solution = 1;
  bool is_success = 2;
}

message WorkerTspStatusUpdate {
  string task_id = 1;
}